src/main/java/mods/railcraft/common/blocks/tracks/TrackSwitch.java

```diff
@@ -44,7 +46,7 @@ public IIcon getIcon() {
     @Override
     public int getBasicRailMetadata(EntityMinecart cart) {
         int meta = tileEntity.getBlockMetadata();
-        if (cart != null && isSwitched()) {
+        if (cart != null && (isSwitched() || springingCarts.contains(cart.getPersistentID()))) {
```

Comment:
Hmm...however, TileSwitchBase I think needs the cosmetic result from isSwitched() in all cases. Maybe don't change isSwitched(), just use the switched variable.

https://github.com/CovertJaguar/Railcraft/blob/master/src/main/java/mods/railcraft/common/blocks/signals/TileSwitchBase.java#L116
https://github.com/CovertJaguar/Railcraft/blob/master/src/main/java/mods/railcraft/common/blocks/signals/TileSwitchBase.java#L190